lib/std/unicode.zig

```diff
@@ -629,3 +629,71 @@ test "utf8ToUtf16LeWithNull" {
         testing.expect(utf16[2] == 0);
     }
 }
+
+pub fn utf8ToUtf16LeStringLiteral(comptime utf8: []const u8) *const [calcUtf16LeLen(utf8) :0] u16 {
+    comptime {
+        const len: usize = calcUtf16LeLen(utf8);
+        var utf16le: [len :0]u16 = [_ :0]u16{0} ** len;
+        const utf16le_len = utf8ToUtf16Le(&utf16le, utf8[0..]) catch |err| @compileError(err);
+        assert(len == utf16le_len);
+        return &utf16le;
+    }
+}
+
+fn calcUtf16LeLen(comptime utf8: []const u8) usize {
+    comptime {
```

Comment:
These two `comptime` keywords are superfluous. Is there any reason this function would not work with a runtime-known parameter?

Reply:
No, it should work with a runtime-known parameter. When it was suggested that a function be used to calculate the length, I thought it was said that it needs to be comptime known and in my mind, that meant comptime keywords. I'll make that change.